logic/tool/TypeResolver.java

```diff
@@ -373,88 +382,72 @@ private void convertValue(TypeVariable owner, ValueConstraint<?> constraint) {
+            else if (constraint.isVariable()) resolvers.register(constraint.asVariable().value());
             else throw GraknException.of(ILLEGAL_STATE);
-            if (noInformation(owner)) addRootTypeVar(owner, rootAttributeType);
+            if (hasNoInfo(owner)) addRootTypeVar(owner, rootAttributeType);
         }
 
-        public void addNeighbours(TypeVariable from, TypeVariable to) {
-            neighbours.putIfAbsent(from, new HashSet<>());
-            neighbours.putIfAbsent(to, new HashSet<>());
-            neighbours.get(from).add(to);
-            neighbours.get(to).add(from);
+        private void addNeighbours(TypeVariable from, TypeVariable to) {
+            neighbours.computeIfAbsent(from, k -> new HashSet<>()).add(to);
+            neighbours.computeIfAbsent(to, k -> new HashSet<>()).add(from);
         }
 
-        public void addNeighboursOfTypeVariable(TypeVariable fromCopy, TypeVariable toCopy) {
-            for (TypeConstraint constraint : fromCopy.constraints()) {
-                if (constraint.isSub()) {
-                    addNeighbours(fromCopy, constraint.asSub().type());
-                } else if (constraint.isOwns()) {
-                    addNeighbours(fromCopy, constraint.asOwns().attribute());
-                } else if (constraint.isPlays()) {
-                    addNeighbours(fromCopy, constraint.asPlays().role());
-                } else if (constraint.isRelates()) {
-                    addNeighbours(fromCopy, constraint.asRelates().role());
-                } else if (constraint.isIs()) {
-                    addNeighbours(fromCopy, constraint.asIs().variable());
-                }
+        private void copyNeighbours(TypeVariable from, TypeVariable to) { // TODO: why is 'to' variable never used?
+            for (TypeConstraint constraint : from.constraints()) {
+                if (constraint.isSub()) addNeighbours(from, constraint.asSub().type());
+                else if (constraint.isOwns()) addNeighbours(from, constraint.asOwns().attribute());
+                else if (constraint.isPlays()) addNeighbours(from, constraint.asPlays().role());
+                else if (constraint.isRelates()) addNeighbours(from, constraint.asRelates().role());
+                else if (constraint.isIs()) addNeighbours(from, constraint.asIs().variable());
+                // TODO: There are other constraints in a TypeVariable. Are we intentionally ignoring them?
+                //       Why do we not throw GraknException.of(ILLEGAL_STATE); ?
             }
         }
-
     }
 
-    private static class ResolverVariables {
+    private static class Resolvers {
 
-        private final Map<Identifier, TypeVariable> resolveeVars;
-        private final Map<RelationConstraint.RolePlayer, TypeVariable> rolePlayerResolveVars;
+        private final Map<Identifier, TypeVariable> typeResolvers;
+        private final Map<RelationConstraint.RolePlayer, TypeVariable> roleTypeResolvers;
```

Comment:
Why is there no checkers and getters for 'roleTypeResolvers'? Do we still need 'roleTypeResolvers' given that we no longer put resolveRoleTypes on RelationConstraints?

Reply:
I've been able to avoid any kind of `roleTypeResolvers` in the new implementation.